cli/lsp/language_server.rs

```diff
@@ -3254,9 +3310,15 @@ impl tower_lsp::LanguageServer for LanguageServer {
       ls.refresh_documents_config().await;
       ls.diagnostics_server.invalidate_all();
       ls.send_diagnostics_update();
-    }
+      ls.outside_lock_task_rx.take().unwrap()
+    };
 
-    lsp_log!("Server ready.");
+    let ls = self.clone();
+    spawn(async move {
+      while let Some(task_fn) = outside_lock_task_rx.recv().await {
+        task_fn(ls.clone()).await;
+      }
+    });
```

Comment:
Alternatively, keep the join handle here and wait for it to complete before finishing the startup sequence.

Reply:
This is an infinite loop, since it's not just done on startup e.g. config changes.